### PR TITLE
Added a LRU Cache file 

### PR DESCRIPTION
**Pull request by Emkay2309**
## Description

Added LRU Cache file and necessary methods to implement in the Cache in Registry

which would act as a memoized data structure , which would always give the least recently used value .
Decreasing the time complexity.

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

**Comment by Emkay2309** [comment]
Requesting for the review of the code.

**Comment by Emkay2309** [comment]
//

**Comment by Emkay2309** [comment]
Added LRU Cache file

File: packages/web3-eth-ens/src/resolver.ts [hunk 1]
```diff
@@ -29,13 +29,48 @@ import { namehash } from './utils.js';
 //  Default public resolver
 //  https://github.com/ensdomains/resolvers/blob/master/contracts/PublicResolver.sol
 
+
+import { LRUCache } from './LRUCache.js';
+
+// Define the cache capacity
+const CACHE_CAPACITY = 100; // You can adjust this number as needed
+const resolverCache = new LRUCache<string, any>(CACHE_CAPACITY);
```
Comment (jdevcs) [inline packages/web3-eth-ens/src/resolver.ts:37]: pls make capacity configurable by users by adding resolverCache in Resolver class and add getter & setter for capacity if any one want to use it. The default should be 100.

File: packages/web3-eth-ens/src/resolver.ts [hunk 2]
```diff
@@ -29,13 +29,48 @@ import { namehash } from './utils.js';
...
+
+	private async fetchENSName(name: string): Promise<any> {
+		// Replace this with your actual network request logic
+		// Example:
+		const response = await fetch(`https://api.example.com/ens/${name}`);
+		if (!response.ok) {
+			throw new Error('Network response was not ok');
+		}
+		return await response.json();
+	}
```
Comment (jdevcs) [inline packages/web3-eth-ens/src/resolver.ts:70]:
```suggestion

```
add cache support for :
getAddress(..) already defined below.

Comment (jdevcs) [reply packages/web3-eth-ens/src/resolver.ts:70]: and also for getName(...) defined below in this class

File: packages/web3-eth-ens/src/LRUCache.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+// src/utils/LRUCache.ts
+
+export class LRUCache<K, V> {
```
Comment (jdevcs) [inline packages/web3-eth-ens/src/LRUCache.ts:3]: pls add unit tests for full coverage of this class

**Comment by jdevcs** [comment]
@Emkay2309

**Review by jdevcs — COMMENTED** [review]
@Emkay2309  closing this PR for : https://github.com/web3/web3.js/issues/7180#issuecomment-2327126698

**Pull request closed without merging**